pkg/backup/backup/backup_cleaner.go

```diff
@@ -49,7 +49,9 @@ func NewBackupCleaner(deps *controller.Dependencies, statusUpdater controller.Ba
 }
 
 func (bc *backupCleaner) Clean(backup *v1alpha1.Backup) error {
-	if backup.DeletionTimestamp == nil || !v1alpha1.IsCleanCandidate(backup) || v1alpha1.NeedNotClean(backup) {
+	if backup.DeletionTimestamp == nil || // Backup CR is created or updated, not deleted
+		!v1alpha1.CleanPolicyIsNotRetain(backup) || // cleanPolicy is CleanPolicyTypeRetain
```

Comment:
Maybe
1. Change func `CleanPolicyIsRetain` to `backup.Spec.CleanPolicy == CleanPolicyTypeRetain` 
2. Remove `!` in here
is cleaner

Reply:
this is inevitable, because there're other usages of just `CleanPolicyIsNotRetain` (yes there's no `!` ahead)